src/imageTools/freehand.js

```diff
@@ -296,9 +296,12 @@ function mouseDownCallback (e, eventData) {
           config.currentTool = toolIndex;
         }
       } else {
-        startDrawing(eventData);
-        addPoint(eventData);
+        if(config.modifying !== 0) {
```

Comment:
Possibly instead of overloading `config.modifying`, we could just add a custom attribute like `config.allowNewROIs` or something similar. All the pipeline for overriding this attribute would appear to be the same. Just it's a bit more explicit.